packages/widgets/src/docklayout.ts

```diff
@@ -343,10 +343,14 @@ export class DockLayout extends Layout {
 
     // Create the root node for the new config.
     if (mainConfig) {
-      this._root = Private.realizeAreaConfig(mainConfig, {
-        createTabBar: () => this._createTabBar(),
-        createHandle: () => this._createHandle()
-      });
+      this._root = Private.realizeAreaConfig(
+        mainConfig,
+        {
+          createTabBar: () => this._createTabBar(),
+          createHandle: () => this._createHandle()
```

Comment:
I'm sorry @GordonSmith I have a doubt due to this intermediate renderer definition. Actually, the `this._document` pass here is not use as these two callbacks are not passing through the received argument (and it does not make sense to allow a document other than `this._document` to be passed).
And as this is the only place `Private.realizeAreaConfig` is called, it sounds that your initial commit only removing `this._document` in `Private.realizeTabAreaConfig` is the only needed change, isn't it?

Reply:
I think its ok as they are just there to satisfy the IRenderer and resolve back onto itself:
```
  private _createTabBar(): TabBar<Widget> {
    // Create the tab bar using the renderer.
    let tabBar = this.renderer.createTabBar(this._document);

...
  }
``` 
IOW changing to this:
``` 
          createTabBar: (document?: Document | SahdowRoot) => this._createTabBar(this._document),
          createHandle: () => this._createHandle()
```
Ends up exactly the same?

You could argue that the following is better from a self-documenting point of view?
``` 
          createTabBar: (_document?: Document | SahdowRoot) => this._createTabBar(),
          createHandle: () => this._createHandle()
```

Reply:
Thanks for the detailed answer. I agree with your latest proposal:

```
          createTabBar: (document?: Document | ShadowRoot) => this._createTabBar(),
          createHandle: () => this._createHandle()
```

It will indeed be a better self-documenting code. I opened https://github.com/jupyterlab/lumino/pull/285 to add this.